cirq-web/cirq_ts/src/circuit/components/types.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export interface Gate {
+  readonly moment: number;
+}
+
+export interface SingleQubitGate extends Gate {
```

Comment:
As discussed, I'd rather move away from the Gate nomenclature here and use something like `OperationRenderer` which takes 
- a list of qubits,
- a list of strings (in the same order as the qubits, which are the labels from circuit_diagram_info, unless overridden in the cirq_web mapping)
- a color 
and returns 
- a list of 3DSymbol objects that can be added to the given qubits